src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +33,14 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                return this.ExpectedValue.GetType().Name;

```

Comment:
Shouldn't the type be prefixed by "Faked" if `ExpectedValue` is a fake? This is what the default `ToString()` implementation does for fakes.